enhancements/network/baremetal-ipi-network-configuration.md

```diff
@@ -0,0 +1,281 @@
+
+In install-config this would look like:
+```yaml
+platform:
+  baremetal:
+    hosts:
+      - name: openshift-master-0
+        networkConfig:
+        - interface: eth0
+          data: |
+            [connection]
+            id=eth0
+            uuid=18e0cec7-041c-4fb3-957c-a60c80dd9b85
+            type=ethernet
+            etc...
+```
+
+Because the initial implementation will be baremetal-specific, we can put the
+network configuration data into the baremetal host field, which will allow easy
+mapping to the machine in question.
+
+### Processing user configuration
+
+#### Deployment of the controlplane hosts via terraform
+
+We will map the keyfiles to their appropriate BareMetalHost using the host field
+of the baremetal install-config. The keyfiles will then be added to custom
+images for each host built by Terraform and Ironic.
+
+Since different configuration may be needed for each host (for example, when
+deploying with static IPs), a Secret per host will be created. A possible
+future optimization is to use a single secret for scenarios such as VLANs
+where multiple hosts can consume the same configuration, but the initial
+implementation will have a 1:1 Secret:BareMetalHost mapping.
+
+#### Deployment of compute/worker hosts
+
+BareMetalHost resources for workers will be created with the Secret containing the network data referenced in the `preprovisioningNetworkData` field defined in the Metal³ [image builder integration design](https://github.com/metal3-io/metal3-docs/blob/master/design/baremetal-operator/image-builder-integration.md#custom-agent-image-controller).
+This will cause the baremetal-operator to create a PreprovisioningImage CRD and wait for it to become available before booting the IPA image.
+
+An OpenShift-specific PreprovisioningImage controller will use the provided network data to build a CoreOS IPA image with the correct ignition configuration in place, and the baremetal-operator will then use this to boot the Host.
+
+If not overridden, the contents of the same network data secret will be passed to the Ironic custom deploy step for CoreOS that installs the node.
+
+### Test Plan
+
+Support will be added to [dev-scripts](https://github.com/openshift-metal3/dev-scripts) for deploying the baremetal network without DHCP enabled. A CI job will populate install-config with the appropriate network configuration and verify that deployment works properly.
+
+### Graduation Criteria
+
+We expect to support this immediately on the baremetal IPI platform.
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+N/A
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+There should be little impact on upgrades and downgrades. Nodes are deployed with network configuration baked into the image, which means it will remain over upgrades or downgrades. NetworkManager keyfiles are considered a stable interface so any version of NetworkManager should be able to parse them equally.
```

Comment:
Should we talk about the whole CoreOS IPA business here? It does have upgrade impact.

Reply:
I think that needs another enhancement proposal.